// @nuxtjs/plausible/dist/module.mjs
import { defu } from 'defu';
import { defineNuxtModule, createResolver, addImports, addPlugin } from '@nuxt/kit';

const name = "@nuxtjs/plausible";
const version = "0.2.4";

const module = defineNuxtModule({
  meta: {
    name,
    version,
    configKey: "plausible",
    compatibility: {
      nuxt: "^3"
    }
  },
  defaults: {
    hashMode: false,
    trackLocalhost: false,
    domain: "",
    apiHost: "https://plausible.io",
    autoPageviews: true,
    autoOutboundTracking: false
  },
  setup(options, nuxt) {
    const { resolve } = createResolver(import.meta.url);
    nuxt.options.runtimeConfig.public.plausible = defu(
      nuxt.options.runtimeConfig.public.plausible,
      options
    );
    nuxt.options.build.transpile.push(resolve("runtime"));
    addImports(
      ["useTrackEvent", "useTrackPageview"].map((name2) => ({
        name: name2,
        as: name2,
        from: resolve(`runtime/composables/${name2}`)
      }))
    );
    addPlugin({
      src: resolve("runtime/plugin.client"),
      mode: "client"
    });
  }
});

export { module as default };
